src/lib/expansion/expansion-panel.ts

```diff
@@ -142,6 +142,7 @@ export class MatExpansionPanel extends _MatExpansionPanelMixinBase
   }
 
   ngOnDestroy() {
+    super.ngOnDestroy();
```

Comment:
This used to be necessary in Angular 2.x, but it should be called automatically in the newer versions.

Reply:
Thanks for the response!

But this does not seem to be the case here, not in my SPA application at least. I am using angular 5.0.1 and material2 5.0.0-rc0 together with angular cli 1.5.0.
And if checking my answer in (#8403) this problem also seems to be existing in the angular material homepage.

I get a Detatched DOM tree for each expansion panel I have disposed. The ngOnDestroy is never called in CdkAccordionItem.

If manually changing in the material2 js generated code, for MatExpansionPanel, in my node_modules folder expansion.js
![image](https://user-images.githubusercontent.com/7062803/32719474-ceee6ffc-c860-11e7-8c06-544354b5c26f.png)

or for es5 expansion.es5.js:
![image](https://user-images.githubusercontent.com/7062803/32719493-dd929448-c860-11e7-9626-7f2656f0f4e7.png)

Then the ngOnDestroy is called in CdkAccordionItem and the issue with Detatched DOM trees are removed.